egs/wsj/asr2/run.sh

```diff
@@ -0,0 +1,370 @@
+        cat ${lmdatadir}/train_trans.txt ${lmdatadir}/train_others.txt > ${lmdatadir}/train.txt
+    fi
+
+    # use only 1 gpu
+    if [ ${ngpu} -gt 1 ]; then
+        echo "LM training does not support multi-gpu. signle gpu will be used."
+    fi
+    ${cuda_cmd} --gpu ${ngpu} ${lmexpdir}/train.log \
+        lm_train.py \
+        --ngpu ${ngpu} \
+        --backend ${backend} \
+        --verbose 1 \
+        --outdir ${lmexpdir} \
+        --tensorboard-dir tensorboard/${lmexpname} \
+        --train-label ${lmdatadir}/train.txt \
+        --valid-label ${lmdatadir}/valid.txt \
+        --test-label ${lmdatadir}/test.txt \
+        --resume ${lm_resume} \
+        --layer ${lm_layers} \
+        --unit ${lm_units} \
+        --opt ${lm_opt} \
+        --sortagrad ${lm_sortagrad} \
+        --batchsize ${lm_batchsize} \
+        --epoch ${lm_epochs} \
+        --patience ${lm_patience} \
+        --maxlen ${lm_maxlen} \
+        --dict ${lmdict}
+fi
+
+
+if [ -z ${tag} ]; then
+    expname=${train_set}_${backend}_${ntype}_${input_layer}_e${elayers}_unit${eunits}_d${dlayers}_unit${dunits}_aheads${aheads}_dim${adim}_mtlalpha${mtlalpha}_${opt}_clip${grad_clip}_sampprob${samp_prob}_ngpu${ngpu}_bs${batchsize}_lr${lr_init}_warmup${warmup_steps}_dropout${dropout}_attndropout${attn_dropout}_mli${maxlen_in}_mlo${maxlen_out}_ninit_${ninit}_epochs${epochs}_accum${accum_grad}_lennorm${len_norm}
+    if [ "${lsm_type}" != "" ]; then
+        expname=${expname}_lsm${lsm_type}${lsm_weight}
+    fi
+    if ${do_delta}; then
+        expname=${expname}_delta
+    fi
+else
+    expname=${train_set}_${backend}_${tag}
+fi
+expdir=exp/${expname}
+mkdir -p ${expdir}
+
+if [ ${stage} -le 4 ] && [ ${stop_stage} -ge 4 ]; then
+    echo "stage 4: Network Training"
+
+    ${cuda_cmd} --gpu ${ngpu} ${expdir}/train.log \
+        asr_train.py \
+        --accum-grad ${accum_grad} \
+        --ngpu ${ngpu} \
+        --backend ${backend} \
+        --outdir ${expdir}/results \
+        --debugmode ${debugmode} \
+        --dict ${dict} \
+        --debugdir ${expdir} \
+        --minibatches ${N} \
+        --verbose ${verbose} \
+        --resume ${resume} \
+        --seed ${seed} \
+        --train-json ${feat_tr_dir}/data.json \
+        --valid-json ${feat_dt_dir}/data.json \
+        --elayers ${elayers} \
+        --eunits ${eunits} \
+        --subsample ${subsample} \
+        --dlayers ${dlayers} \
+        --dunits ${dunits} \
+        --aheads ${aheads} \
+        --adim ${adim} \
+        --mtlalpha ${mtlalpha} \
+        --batch-size ${batchsize} \
+        --dropout-rate ${dropout} \
+        --maxlen-in ${maxlen_in} \
+        --maxlen-out ${maxlen_out} \
+        --opt ${opt} \
+        --grad-clip ${grad_clip} \
+        --sampling-probability ${samp_prob} \
+        --epochs ${epochs} \
+        --sortagrad ${sortagrad} \
+        --lsm-weight ${lsm_weight} \
+        --model-module "espnet.nets.${backend}_backend.asr_transformer" \
```

Comment:
It is not user-friendly to assume implicitly that `E2E` class is defined in this module. How about specifying explicitly `espnet.nets.${backend}_backend.asr_transformer.E2E`  instead?